proxysql_tools/managers/galera_manager.py

```diff
@@ -70,33 +80,51 @@ def discover_cluster_nodes(self):
                         'password': initial_node.password
                     })
 
-                    try:
-                        node.refresh_state()
-                        if not node.cluster_status == CLUSTER_STATUS_PRIMARY:
-                            raise GaleraNodeNonPrimary(
-                                'Node: %s:%s is in non-primary state' %
-                                (node.host, node.port)
-                            )
-                    except ModelValidationError as e:
-                        # The node state cannot be refreshed as some of the
-                        # properties of the node could not be fetched. We
-                        # should fail the discovery in such a case as this is
-                        # unexpected error.
-                        raise GaleraNodeUnknownState(e.messages)
+                    # Check that the initial node status is 'PRIMARY'
+                    self.refresh_and_validate_node_state(node)
 
                     if GaleraManager.nodes_in_same_cluster(initial_node, node):
                         self._nodes.append(node)
 
         return True
 
     @staticmethod
+    def refresh_and_validate_node_state(node):
+        """Validates the state of the node to ensure that the node's status
+        is PRIMARY and that all the node's properties can be fetched.
+
+        :param node: The object that stores information on the Galera node.
+        :type node: GaleraNode
+        """
+        try:
+            node.refresh_state()
+            if not node.cluster_status == CLUSTER_STATUS_PRIMARY:
+                err_msg = ('Node %s:%s is in non-primary '
+                           'state %s' % (node.host, node.port,
+                                         node.cluster_status))
+
+                log.error(err_msg)
+                raise GaleraNodeNonPrimary(err_msg)
+        except ModelValidationError as e:
```

Comment:
I see you expect `node.refresh_state()` to raise `ModelValidationError` while the doc says nothing about the exception. We should either handle the exceptions in `refresh_state()` or add the exceptions to docstring.